feature/foryou/src/main/java/com/google/samples/apps/nowinandroid/feature/foryou/ForYouViewModel.kt

```diff
@@ -148,15 +160,20 @@ class ForYouViewModel @Inject constructor(
 
     val interestsSelectionState: StateFlow<ForYouInterestsSelectionUiState> =
         combine(
+            syncStatusMonitor.isSyncing,
             followedInterestsUiState,
             topicsRepository.getTopicsStream(),
             authorsRepository.getAuthorsStream(),
-            snapshotFlow { inProgressTopicSelection },
-            snapshotFlow { inProgressAuthorSelection },
-        ) { followedInterestsUserState, availableTopics, availableAuthors, inProgressTopicSelection,
-            inProgressAuthorSelection ->
+            combine(
+                snapshotFlow { inProgressTopicSelection },
+                snapshotFlow { inProgressAuthorSelection },
+                ::Pair,
+            )
+        ) { isSyncing, followedInterestsUserState, availableTopics, availableAuthors,
+            (inProgressTopicSelection, inProgressAuthorSelection) ->
 
-            when (followedInterestsUserState) {
+            if (isSyncing) ForYouInterestsSelectionUiState.Loading
```

Comment:
I'm slightly concerned about this logic.

If we currently have data, shouldn't we be showing that, even if we are syncing in the background?

Reply:
Great point, I did want to show feedback to the user when loading even though there's data however. I've added a `isLoading` field for `ForYouInterestsSelectionUiState.WithInterestsSelection` to achieve this.

Reply:
I ended up exposing the `isSyncing` field separately and de-duping the loading spinners in the interests and news components. The loading spinner is now owned by the `ForYouScreen` (and subsequently the `BookmarksScreen`) as a whole.

The spinner now shows:
On the `ForYouScreen` if sync is in progress or either interests or news are loading.
On the `BookmarksScreen` is news are loading. A future PR can neatly establish loading if sync is in progress here as well.